.github/workflows/_test_slurm_pyxis.yaml

```diff
@@ -0,0 +1,248 @@
+    outputs:
+      SLURM_JOB_ID:
+        description: ID of the SLURM job
+        value: ${{ jobs.run-test.outputs.SLURM_JOB_ID }}
+      SLURM_STATE:
+        description: State of the SLURM job
+        value: ${{ jobs.run-test.outputs.SLURM_STATE }}
+      SLURM_EXITCODE:
+        description: Exit code of the SLURM job
+        value: ${{ jobs.run-test.outputs.SLURM_EXITCODE }}
+
+jobs:
+
+  run-test:
+    name: ${{ inputs.NAME }}
+    runs-on: ubuntu-22.04
+    outputs:
+      SLURM_JOB_ID: ${{ steps.submit.outputs.SLURM_JOB_ID }}
+      SLURM_STATE: ${{ steps.exit-info.outputs.SLURM_STATE }}
+      SLURM_EXITCODE: ${{ steps.exit-info.outputs.SLURM_EXITCODE }}
+    steps:
+      - name: Print environment variables
+        run: env
+
+      - name: Check out the repository under ${GITHUB_WORKSPACE}
+        uses: actions/checkout@v4
+
+      - name: Setup SSH agent
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      - name: Setup SSH known hosts
+        id: ssh-known-hosts
+        run: |
+          mkdir -p ~/.ssh
+          cat >> ~/.ssh/known_hosts << EOF
+          ${{ vars.SSH_KNOWN_HOSTS }}
+          EOF
+          chmod 600 ~/.ssh/known_hosts
+          echo "FILE=$(realpath ~/.ssh/known_hosts)" >> $GITHUB_OUTPUT
+
+      - name: Labels and metadata
+        id: meta
+        shell: bash -x -e {0}
+        run: |
+          IMAGE="$(echo ${{inputs.IMAGE}} | sed 's/\//#/')"
+          ARTIFACT_PATH=${{ inputs.SLURM_SCRATCH_PATH }}/${{ github.run_id }}/${{ inputs.NAME }}
+          LOG_FILE=${{ inputs.SLURM_SCRATCH_PATH }}/${{ github.run_id }}/${{ inputs.NAME }}.log
+          for var in IMAGE LOG_FILE ARTIFACT_PATH; do
+            echo "$var=${!var}" >> $GITHUB_OUTPUT
+          done
+          
+      - name: Submit SLURM jobs over SSH
+        id: submit
+        shell: bash -O expand_aliases -x -e {0}
+        run: |
+          alias SSH='ssh ${{ secrets.SLURM_LOGIN_USER }}@${{ inputs.SLURM_LOGIN_HOSTNAME }}'
+          SSH mkdir -p "${{ steps.meta.outputs.ARTIFACT_PATH }}"
+          SLURM_JOB_ID=$(SSH sbatch --parsable <<"EOF"
+          #!/bin/bash
+          #SBATCH --job-name=${{ github.run_id }}-${{ inputs.NAME }}
+          #SBATCH --exclusive
+          #SBATCH --nodes=${{ inputs.NODES }}
+          #SBATCH --gpus-per-node=${{ inputs.GPUS_PER_NODE }}
+          #SBATCH --time=${{ inputs.TIME_LIMIT }}
+          #SBATCH --output=${{ steps.meta.outputs.LOG_FILE }}
+          #SBATCH --export="${{ inputs.EXTRA_EXPORTS }},ENROOT_PASSWORD=${{ secrets.CONTAINER_REGISTRY_TOKEN }}"
+
+          # preload enroot container using one task per node
+          time srun \
+            --ntasks-per-node=1 \
+            --container-name=runtime \
+            --container-image=${{ steps.meta.outputs.IMAGE }} \
+            true
+            
+          # run single-task preambles for, e.g., dependencies installation
+          time srun \
```

Comment:
I'm not sure how srun+pyxis work, so I have a question: will the changes remain in the docker container after completion of this cmd?

Reply:
See [this comment](https://github.com/NVIDIA/JAX-Toolbox/pull/619#discussion_r1514960392).